homeassistant/components/nibe_heatpump/climate.py

```diff
@@ -0,0 +1,349 @@
+        cooling_with_room_sensor_address=ADDRESS_COOLING_WITH_ROOM_SENSOR_F,
+    ),
+    ClimateDescription(
+        key="s4",
+        name="Climate System S4",
+        current_address=40030,
+        setpoint_heat_address=47395,
+        setpoint_cool_address=48782,
+        prio_address=ADDRESS_PRIO_F,
+        mixing_valve_state_address=43093,
+        active_accessory_address=47304,
+        use_room_sensor_address=47391,
+        cooling_with_room_sensor_address=ADDRESS_COOLING_WITH_ROOM_SENSOR_F,
+    ),
+)
+
+CLIMATE_SYSTEMS_S = (
+    ClimateDescription(
+        key="s1",
+        name="Climate System S1",
+        current_address=30027,
+        setpoint_heat_address=40207,
+        setpoint_cool_address=40989,
+        prio_address=ADDRESS_PRIO_S,
+        mixing_valve_state_address=31034,
+        active_accessory_address=None,
+        use_room_sensor_address=40203,
+        cooling_with_room_sensor_address=ADDRESS_COOLING_WITH_ROOM_SENSOR_S,
+    ),
+    ClimateDescription(
+        key="s2",
+        name="Climate System S2",
+        current_address=30026,
+        setpoint_heat_address=40206,
+        setpoint_cool_address=40988,
+        prio_address=ADDRESS_PRIO_S,
+        mixing_valve_state_address=31033,
+        active_accessory_address=None,
+        use_room_sensor_address=40202,
+        cooling_with_room_sensor_address=ADDRESS_COOLING_WITH_ROOM_SENSOR_S,
+    ),
+    ClimateDescription(
+        key="s3",
+        name="Climate System S3",
+        current_address=30025,
+        setpoint_heat_address=40205,
+        setpoint_cool_address=40987,
+        prio_address=ADDRESS_PRIO_S,
+        mixing_valve_state_address=31032,
+        active_accessory_address=None,
+        use_room_sensor_address=40201,
+        cooling_with_room_sensor_address=ADDRESS_COOLING_WITH_ROOM_SENSOR_S,
+    ),
+    ClimateDescription(
+        key="s4",
+        name="Climate System S4",
+        current_address=30024,
+        setpoint_heat_address=40204,
+        setpoint_cool_address=40986,
+        prio_address=ADDRESS_PRIO_S,
+        mixing_valve_state_address=31031,
+        active_accessory_address=None,
+        use_room_sensor_address=40200,
+        cooling_with_room_sensor_address=ADDRESS_COOLING_WITH_ROOM_SENSOR_S,
+    ),
+)
+
+CLIMATE_SYSTEMS = {Series.F: CLIMATE_SYSTEMS_F, Series.S: CLIMATE_SYSTEMS_S}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up platform."""
+
+    coordinator: Coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    def climate_systems():
+        for description in CLIMATE_SYSTEMS.get(coordinator.series, ()):
+            try:
+                yield Climate(coordinator, description)
+            except CoilNotFoundException as exception:
+                LOGGER.debug(
+                    "Skipping climate: %s due to %s", description.key, exception
+                )
```

Comment:
Is this expected to happen under normal circumstances or should it be a higher log level?

Reply:
It is expected to happen, but it's not an issue. Some pumps define many of these climate systems some don't.